app/handlers/newflow/educator_complete_profile.rb

```diff
@@ -0,0 +1,81 @@
+module Newflow
+  class EducatorCompleteProfile
+    lev_handler
+
+    paramify :signup do
+      attribute :educator_specific_role, type: String
+      attribute :other_role_name, type: String
+      attribute :how_are_books_chosen, type: String
+      attribute :using_openstax_how, type: String
+      attribute :num_students_per_semester_taught, type: Integer
+      attribute :subjects_of_interest, type: Object
+      attribute :books_used, type: Object
+
+      validates(
+        :educator_specific_role,
+        inclusion: {
+          in: %w(instructor administrator other),
+        }
+      )
+
+      validates(
+        :num_students_per_semester_taught,
+        numericality: {
+          only_integer: true,
+        }
+      )
+    end
+
+    protected ###############
+
+    def authorized?
+      !caller.is_anonymous?
+    end
+
+    def handle
+      caller.update(state: User::EDUCATOR_COMPLETE_PROFILE)
+      transfer_errors_from(caller, {type: :verbatim}, true)
+
+      push_lead_to_salesforce
```

Comment:
what happens if this fails?  e.g., salesforce is down?  how would your team know of a problem?  is the call logged somewhere in case someone wants to know how many leads have failed/succeeded in pushing from accounts to SF?

Reply:
The object that handles that takes care of it. I believe it is intentional to allow users to continue with the flow. https://github.com/openstax/accounts/blob/f2de3c2f17a8f613bf9a99667dcd2448942f4fb9/app/routines/push_salesforce_lead.rb#L67-L77